packages/dds/merge-tree/src/perspective.ts

```diff
@@ -0,0 +1,239 @@
+				)
+			) {
+				next = seg;
+				return LeafAction.Exit;
+			}
+		};
+		(forward ? forwardExcursion : backwardExcursion)(segment, action);
+		return next ?? (forward ? this._mergeTree.endOfTree : this._mergeTree.startOfTree);
+	}
+
+	/**
+	 * @param segment - The segment to start from.
+	 * @returns the previous segment, or the start of the tree if there is no previous segment.
+	 * @remarks This is a convenient equivalent to calling `nextSegment(segment, false)`.
+	 */
+	public previousSegment(segment: ISegment): ISegment {
+		return this.nextSegment(segment, false);
+	}
+}
+
+/**
+ * @param seg - The segment to check.
+ * @param seq - The latest sequence number to consider.
+ * @param clientId - The ID of the client to consider.
+ * @param refSeq - The reference sequence number,
+ * aka the latest sequence number to consider for other clients' changes.
+ * @param localSeq - The latest local sequence number to consider.
+ * @returns true iff this segment was removed in the given perspective.
+ */
+export function wasRemovedBefore(
+	seg: ISegment,
+	seq: number,
+	clientId: number,
+	refSeq: number,
+	localSeq?: number,
+): boolean {
+	if (
+		seg.removedSeq === UnassignedSequenceNumber &&
+		localSeq !== undefined &&
+		seg.localRemovedSeq !== undefined
+	) {
+		return seg.localRemovedSeq <= localSeq;
+	}
+	if (seg.removedClientIds !== undefined && seg.removedClientIds.includes(clientId)) {
+		return (
+			seg.removedSeq !== undefined &&
+			seg.removedSeq !== UnassignedSequenceNumber &&
+			seg.removedSeq <= seq
+		);
+	}
+	return (
+		seg.removedSeq !== undefined &&
+		seg.removedSeq !== UnassignedSequenceNumber &&
+		seg.removedSeq <= refSeq
+	);
+}
+
+/**
+ * @param seg - The segment to check.
+ * @param seq - The latest sequence number to consider.
+ * @param clientId - The ID of the client to consider.
+ * @param refSeq - The reference sequence number,
+ * aka the latest sequence number to consider for other clients' changes.
+ * @param localSeq - The latest local sequence number to consider.
+ * @returns true iff this segment was moved (aka obliterated) in the given perspective.
+ */
+export function wasMovedBefore(
+	seg: ISegment,
+	seq: number,
+	clientId: number,
+	refSeq: number,
+	localSeq?: number,
+): boolean {
+	if (
+		seg.movedSeq === UnassignedSequenceNumber &&
+		localSeq !== undefined &&
+		seg.localMovedSeq !== undefined
+	) {
+		return seg.localMovedSeq <= localSeq;
+	}
+	if (seg.movedClientIds !== undefined && seg.movedClientIds.includes(clientId)) {
+		return (
+			seg.movedSeq !== undefined &&
+			seg.movedSeq !== UnassignedSequenceNumber &&
+			seg.movedSeq <= seq
+		);
+	}
+	return (
+		seg.movedSeq !== undefined &&
+		seg.movedSeq !== UnassignedSequenceNumber &&
+		seg.movedSeq <= refSeq
+	);
+}
+
+/**
+ * See {@link wasRemovedBefore} and {@link wasMovedBefore}.
+ */
+export function wasRemovedOrMovedBefore(
+	seg: ISegment,
+	seq: number,
+	clientId: number,
+	refSeq: number,
+	localSeq?: number,
```

Comment:
we use this set of values alot. if we had an interface, we could reduce the bundle impact here by just passing around an object

Reply:
Added a `SeqTime` interface. It only includes `refSeq` and `localSeq` as we discussed in standup.